code/__DEFINES/xeno.dm

```diff
@@ -144,3 +144,20 @@ GLOBAL_LIST_INIT(xeno_ai_spawnable, list(
 	} else { \
 		xeno.remove_filter("overheal_vis"); \
 	}
+
+/// Used by the isValidForResinStructure proc.
+/// 1 is reserved for TRUE , 0 for FALSE
+/// Simply not allowed by the area to build
+#define ERROR_NOT_ALLOWED 2
+/// No weeds here, but it is weedable
```

Comment:
I am gonna be picky again, but these are rules existing in other languages, and they are good for maintainability.

A function should return one type. In dm, it happens that booleans do not really exist so we defined them using int. However, we consider them as bool. 

In that sense, your function sometimes return an enum (int) or a bool. That is confusing.
More so, if your function can have different error values that you have to handle, you want the zero code to be the absence of errors; else you have a sucess value for 1, and errors for 0 and 2. Which means you cannot use the truthy value to see if the function fails or not.

Your functio  should return only an enum of int, and int =0 should represent success